WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -0,0 +1,201 @@
+    
+    public override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let count = themesController.fetchedObjects?.count
+        return count ?? 0
+    }
+    
+    public override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> ThemeBrowserCell
+    {
+        let cell = collectionView.dequeueReusableCellWithReuseIdentifier("ThemeBrowserCell", forIndexPath: indexPath) as! ThemeBrowserCell
+        let theme = themesController.objectAtIndexPath(indexPath) as? Theme
+        
+        cell.configureWithTheme(theme)
+        
+        return cell
+    }
+    
+    public override func collectionView(collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, atIndexPath indexPath: NSIndexPath) -> UICollectionReusableView {
+        let header = collectionView.dequeueReusableSupplementaryViewOfKind(kind, withReuseIdentifier: "ThemeBrowserHeaderView", forIndexPath: indexPath)
+        return header
+    }
+    
+    public override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    // MARK: - UICollectionViewController protocol UICollectionViewDelegate
+
+    public override func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
+        
+        if let theme = themesController.objectAtIndexPath(indexPath) as? Theme {
+            showDemoForTheme(theme)
+        }
+    }
+    
+    // MARK: - UICollectionViewDelegateFlowLayout
+    
+    public func collectionView(collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAtIndexPath indexPath: NSIndexPath) -> CGSize {
+        let parentViewWidth = collectionView.frame.size.width
+        let width = cellWidthForFrameWidth(parentViewWidth)
+        
+        return CGSize(width: width, height: ThemeBrowserCell.heightForWidth(width))
+    }
+    
+    // MARK: - Layout calculation helper methods
+    
+    /**
+     *  @brief      Calculates the cell width for parent frame
+     *
+     *  @param      parentViewWidth     The width of the parent view.
+     *
+     *  @returns    The requested cell width.
+     */
+    private func cellWidthForFrameWidth(parentViewWidth : CGFloat) -> CGFloat {
+        let numberOfColumns = max(1, trunc(parentViewWidth / WPStyleGuide.Themes.minimumColumnWidth))
+        let numberOfMargins = numberOfColumns + 1
+        let marginsWidth = numberOfMargins * WPStyleGuide.Themes.columnMargin
+        let columnsWidth = parentViewWidth - marginsWidth
+        let columnWidth = trunc(columnsWidth / numberOfColumns)
+        
+        return columnWidth
+    }
+    
+    // MARK: - UISearchBarDelegate
+    
+    public func searchBar(searchBar: UISearchBar, textDidChange searchText: String) {
+        // SEARCH AWAY!!!
```

Comment:
(I may be missing something) but... do we need to wire something? (is this one a placeholder?)